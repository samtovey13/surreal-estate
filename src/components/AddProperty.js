import React, {useState} from "react";
import "../styles/AddProperty.css";
import createProperty from "../requests/createProperty";

import Alert from './Alert';
import AddPropertyForm from './AddPropertyForm';

const AddProperty = () => {
  const initialState = {
    fields: {
      title: "",
      type: "Flat",
      bedrooms: "",
      bathrooms: "",
      price: "",
      city: "Manchester",
      email: "",
    },
    alert: {
      message: "testing error/success message",
      isSuccess: false,
    },
  };
  const [fields, setFields] = useState(initialState.fields);
  const [alert, setAlert] = useState(initialState.alert);

  const handleFieldChange = (event) => {
    setFields({ ...fields, [event.target.name]: event.target.value });
  };

  const handleAddProperty = async (event) => {
    event.preventDefault();
    setAlert({
      message: "",
      isSuccess: false,
    });
    const res = await createProperty(fields);
    if (res.status === 201) {
      setAlert({
        message: "Success! Property added.",
        isSuccess: true,
      });
    } else {
      setAlert({
        message: "Oops! Something went wrong. Property couldn't be added.",
        isSuccess: false,
      });
    }
  };
    
  return (
    <div className="add-property-container">

    <div className="add-property">
      {alert.message && (
        <Alert message={alert.message} success={alert.isSuccess} />
      )}
      <AddPropertyForm
        fields={fields}
        handleFieldChange={handleFieldChange}
        handleSubmit={handleAddProperty}
      />
    </div></div>
  );
};

export default AddProperty;
