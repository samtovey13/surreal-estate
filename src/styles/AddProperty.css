.add-property-container {
  margin: 0;
  padding: 10px;
}
