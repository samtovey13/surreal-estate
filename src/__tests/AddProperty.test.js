import { render, fireEvent } from "@testing-library/react";
import AddProperty from '../components/AddProperty';

describe("AddProperty", () => {
  const mockFields = {
    title: "",
    type: "Flat",
    bedrooms: "",
    bathrooms: "",
    price: "",
    city: "Manchester",
    email: "",
  };
  const onSubmit = jest.fn();
  const onChange = jest.fn();

  it("renders correctly", () => {
    const { asFragment } = render(
      <AddProperty
        fields={mockFields}
        handleFieldChange={onChange}
        handleSubmit={onSubmit}
      />
    );
    expect(asFragment).toMatchSnapshot();
  })

  it("calls handleSubmit when form is submitted", () => {
    const { getByTestId } = render(
      <AddProperty
        fields={mockFields}
        handleFieldChange={onChange}
        handleSubmit={onSubmit}
      />
    );
    const formElement = getByTestId("add-property-form");

    fireEvent.submit(formElement);
    expect(onSubmit).toHaveBeenCalledTimes(1);
  });
})
