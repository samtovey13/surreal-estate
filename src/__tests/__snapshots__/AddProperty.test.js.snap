// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty renders correctly 1`] = `
<DocumentFragment>
  <div
    class="add-property-container"
  >
    <div
      class="add-property"
    >
      <form
        data-testid="add-property-form"
      >
        <ul
          class="form-ul"
        >
          <li>
            <label
              for="title"
            >
              Title:
            </label>
            <input
              id="title"
              name="title"
              placeholder="Brief description"
              type="text"
              value=""
            />
          </li>
          <li>
            <label
              for="type"
            >
              Type:
            </label>
            <select
              id="type"
              name="type"
            >
              <option
                value="Flat"
              >
                Flat
              </option>
              <option
                value="Semi-Detached"
              >
                Semi-Detached
              </option>
              <option
                value="Terraced"
              >
                Terraced
              </option>
              <option
                value="End of Terrace"
              >
                End of Terrace
              </option>
              <option
                value="Cottage"
              >
                Cottage
              </option>
              <option
                value="Bungalow"
              >
                Bungalow
              </option>
            </select>
          </li>
          <span
            class="bedroom-bathroom-flex-container"
          >
            <li
              class="bedroom-li"
            >
              <label
                for="bedrooms"
              >
                Bedrooms:
              </label>
              <input
                id="bedrooms"
                max="100"
                min="0"
                name="bedrooms"
                placeholder="e.g. 2"
                type="number"
                value=""
              />
            </li>
            <li
              class="bathroom-li"
            >
              <label
                for="bathrooms"
              >
                Bathrooms:
              </label>
              <input
                id="bathrooms"
                max="20"
                min="0"
                name="bathrooms"
                placeholder="e.g. 1"
                type="number"
                value=""
              />
            </li>
          </span>
          <li>
            <label
              for="price"
            >
              Price:
            </label>
            <div
              class="currency-wrapper"
            >
              <input
                id="price"
                min="0"
                name="price"
                placeholder="250,000"
                type="number"
                value=""
              />
              <span
                class="currency-symbol"
              >
                Â£
              </span>
            </div>
          </li>
          <li>
            <label
              for="city"
            >
              City:
            </label>
            <select
              id="city"
              name="city"
              type="select"
            >
              <option
                value="Manchester"
              >
                Manchester
              </option>
              <option
                value="Leeds"
              >
                Leeds
              </option>
              <option
                value="Sheffield"
              >
                Sheffield
              </option>
              <option
                value="Liverpool"
              >
                Liverpool
              </option>
            </select>
          </li>
          <li>
            <label
              for="email"
            >
              Email:
            </label>
            <input
              id="email"
              name="email"
              placeholder="Main contact"
              type="email"
              value=""
            />
          </li>
          <li>
            <button
              class="add-property-button"
              type="submit"
            >
              Add Property
            </button>
          </li>
        </ul>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
